lib/instance.js

```diff
@@ -300,25 +298,23 @@ module.exports = (function() {
             }
 
             // Convert date fields to real date objects
-            if (this.Model._hasDateAttributes && this.Model._isDateAttribute(key) && value !== null && value !== undefined && !(value instanceof Date) && !value._isSequelizeMethod) {
+            if (this.Model._hasDateAttributes && this.Model._isDateAttribute(key) && value && !(value instanceof Date) && !value._isSequelizeMethod) {
               value = new Date(value);
             }
           }
 
           // Convert boolean-ish values to booleans
-          if (this.Model._hasBooleanAttributes && this.Model._isBooleanAttribute(key) && value !== null && value !== undefined && !value._isSequelizeMethod) {
+          if (this.Model._hasBooleanAttributes && this.Model._isBooleanAttribute(key) && value && !value._isSequelizeMethod) {
             if (Buffer.isBuffer(value) && value.length === 1) {
               // Bit fields are returned as buffers
               value = value[0];
             }
 
-            if (_.isString(value)) {
-              // Only take action on valid boolean strings.
-              value = (value === 'true') ? true : (value === 'false') ? false : value;
-
-            } else if (_.isNumber(value)) {
-              // Only take action on valid boolean integers.
-              value = (value === 1) ? true : (value === 0) ? false : value;
+            if (typeof value === 'string' || typeof value === 'number') {
```

Comment:
Well for one the previous code read a ton better, not that it read perfectly though.
Really this parsing should probably happen higher up the change since it's dialect specific boolean parsing.